### PR TITLE
feat(transformer): support `targets` option of preset-env

### PR DESCRIPTION
**Pull request by Dunqing**
The implementation of the `targets` options was copied from [swc](https://github.com/swc-project/swc/tree/main/crates/preset_env_base), which resulted in some added dependencies in transformer.
Currently, it has supported enabling plugins by `targets`

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#3371** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3371?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/Cargo.toml [hunk 1]
```diff
@@ -26,11 +26,18 @@ oxc_diagnostics = { workspace = true }
...
+serde           = { workspace = true, features = ["derive"] }
+serde_json      = { workspace = true }
+ropey           = { workspace = true }
+rustc-hash      = { workspace = true }
+once_cell       = { workspace = true }
+st-map          = { version = "0.2.3" }
+semver          = { version = "1.0.23" }
+anyhow          = { version = "1.0.86" }
+tracing         = { version = "0.1.40" }
+browserslist-rs = { version = "0.15.0", features = ["wasm_bindgen"] }
```
Comment (Boshen) [inline crates/oxc_transformer/Cargo.toml:40]: do we need wasm bindgen?

Comment (Dunqing) [reply crates/oxc_transformer/Cargo.toml:40]: Yes, oxc-wasm needs

Comment (Boshen) [reply crates/oxc_transformer/Cargo.toml:40]: Need to remove `st-map`, `semver`, `tracing`

File: crates/oxc_transformer/src/options/transformer_options.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        })
+    }
+}
+
+fn get_options<T: Default + DeserializeOwned>(
+    name: &str,
+    babel_options: &BabelOptions,
+    errors: &mut Vec<Error>,
+    is_preset: bool,
+) -> T {
```
Comment (Boshen) [inline crates/oxc_transformer/src/options/transformer_options.rs:121]: Are we able to remove this `T`?

Comment (Dunqing) [reply crates/oxc_transformer/src/options/transformer_options.rs]: Now we'll replace it with this code. Do you mind that this needs to write more code?

```rust
let plugin_name = "transform-arrow-functions";
enable_plugin(plugin_name, options, &env_options, &targets).map(|options| {
    from_value::<ArrowFunctionsOptions>(options).unwrap_or_else(|err| {
        report_error(plugin_name, &err, false, &mut errors);
        ArrowFunctionsOptions::default()
    })
})
```

File: crates/oxc_transformer/Cargo.toml (hunk 1 above)
Comment (Dunqing) [reply crates/oxc_transformer/Cargo.toml:36]: All of these are removed!

**Pull request merged**